### PR TITLE
[llbuildSwift] Expose error() delegate callback in Core bindings

### PR DESCRIPTION
**Pull request by dmbryson**
rdar://problem/45728573

### CODE REVIEW

**Comment by dmbryson** [comment]
@swift-ci please smoke test

File: products/llbuildSwift/CoreBindings.swift [hunk 1]
```diff
@@ -237,6 +237,18 @@ public protocol BuildEngineDelegate {
...
+
+    /// Handle error message reported by the engine
+    ///
+    /// The delegate should handle this, but a default implementation is
+    /// is provided that does nothing for backward compatiblity.
+    func error(_ message: String)
+}
+
+extension BuildEngineDelegate {
+    func error(_ message: String) {
```
Comment (aciidb0mb3r) [inline products/llbuildSwift/CoreBindings.swift]: Shouldn't this be public?

File: products/llbuildSwift/CoreBindings.swift [hunk 2]
```diff
@@ -237,6 +237,18 @@ public protocol BuildEngineDelegate {
...
+        return
```
Comment (aciidb0mb3r) [inline products/llbuildSwift/CoreBindings.swift]: nit: not required

File: products/llbuildSwift/CoreBindings.swift (hunk 1 above)
Comment (dmbryson) [reply products/llbuildSwift/CoreBindings.swift]: Probably.  Smoke tests seem to indicate so.  Wonder why my local tests passed? 🤔

File: products/llbuildSwift/CoreBindings.swift (hunk 2 above)
Comment (dmbryson) [reply products/llbuildSwift/CoreBindings.swift]: Is it not?  I don't want to break SwiftPM downstream by merging this with a newly required method.

Comment (aciidb0mb3r) [reply products/llbuildSwift/CoreBindings.swift]: Sorry, I meant the `return` keyword is not required 😅

Comment (dmbryson) [reply products/llbuildSwift/CoreBindings.swift]: Ahh, that makes sense.

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Pull request merged**